src/components/range-inputs.tsx

```diff
@@ -0,0 +1,303 @@
+                            defaultValue={value} pattern='\d{4}-\d{2}-\d{2}' min='1970-01-01' max='2999-12-31' onChange={changeHandler} />
+                }
+                <ClearInputBtn
+                    btnClassName='range-input-clear'
+                    clickCallback={clearInput}
+                    show
+                />
+            </div>
+        )
+
+    )
+};
+
+
+type RangeInputHOCProps = {
+    /**
+     * an string for the type of input expected:
+     * int
+     * float
+     * date
+     * timestamp
+     */
+    inputType: string,
+    /**
+     * use for applying custom styles on the component
+     */
+    classes?: string
+};
+
+const getType = (inputType: string): string => {
+
+    if (inputType.indexOf('int') > -1) return 'int';
+
+    if (inputType.indexOf('float') > -1) return 'float';
+
+    if (inputType.indexOf('date') > -1) return 'date';
+
+    if (inputType.indexOf('timestamp') > -1) return 'timestamp';
+
+    return 'Invalid_Type';
+}
+
+const RangeInputHOC = ({ inputType, classes }: RangeInputHOCProps) => {
+    const momentJS = windowRef.moment;
+
+    const fromRef = useRef<HTMLInputElement>(null);
+
+    const toRef = useRef<HTMLInputElement>(null);
+
+    const fromTimeRef = useRef<HTMLInputElement>(null);
+
+    const toTimeRef = useRef<HTMLInputElement>(null);
+
+    const type = getType(inputType);
+
+    const [error, setError] = useState<string | null>(null);
+
+    const [disableSubmit, setDisableSubmit] = useState<boolean>(type === 'int' || type === 'float');
+
+    const disableSubmitBtn = () => {
+        if (!disableSubmit) setDisableSubmit(true);
+    }
+
+    const enableSubmitBtn = () => {
+        if (disableSubmit) setDisableSubmit(false);
+    }
+
+    const formatTimeValues = () => {
+        const fromDateVal = fromRef?.current?.value;
+        let fromTimeVal = fromTimeRef?.current?.value;
+
+        if (fromDateVal && !fromTimeVal) fromTimeVal = '00:00';
+
+        const toDateVal = toRef?.current?.value;
+        let toTimeVal = toTimeRef?.current?.value;
+
+        if (toDateVal && !toTimeVal) toTimeVal = '00:00';
+
+        return {
+            fromVal: fromDateVal || fromTimeVal ? `${fromDateVal}T${fromTimeVal}` : '',
+            toVal: toDateVal || toTimeVal ? `${toDateVal}T${toTimeVal}` : '',
+        };
+    }
+
+    /**checks if both inputs are empty and disables the submit button */
+    const handleChange = (value: string) => {
+        const formatedValues = type === 'timestamp' ? formatTimeValues()
+            : { fromVal: fromRef?.current?.value || '', toVal: toRef?.current?.value || '' };
+
+        console.log(formatedValues);
+
+        if (!formatedValues.fromVal && !formatedValues.toVal) disableSubmitBtn();
+        else enableSubmitBtn();
+
+        const validatedResult = validateValue(value);
+
+        if (!validatedResult) setError(errorMsgMap[type]);
+
+        if (validatedResult && !error) setError(null);
```

Comment:
One of my comments in slack was if the value was invalid before (error is defined) and it is now valid, we want to hide the error. This check should change to make sure no value is invalid for either `to` or `from` before hiding the error.

Reply:
`      
const validatedResult = validateValue(formatedValues.fromVal) && validateValue(formatedValues.toVal);

if (!validatedResult) setError(errorMsgMap[type])
else setError(null);
`

changed the validation logic check to this so your issue should be resolved now